sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/AsyncRestProxy.java

```diff
@@ -176,12 +178,17 @@ static Mono<?> handleBodyReturnType(HttpResponse sourceResponse, Function<byte[]
             // Mono<Flux<ByteBuffer>>
             asyncResult = Mono.just(sourceResponse.getBody());
         } else if (TypeUtil.isTypeOrSubTypeOf(entityType, BinaryData.class)) {
+            HttpHeader httpHeader = sourceResponse.getHeaders().get(HttpHeaderName.fromString("Content-Type"));
             // Mono<BinaryData>
             // The raw response is directly used to create an instance of BinaryData which then provides
             // different methods to read the response. The reading of the response is delayed until BinaryData
             // is read and depending on which format the content is converted into, the response is not necessarily
             // fully copied into memory resulting in lesser overall memory usage.
-            asyncResult = BinaryData.fromFlux(sourceResponse.getBody());
+            if (httpHeader != null && httpHeader.getValue() != null && httpHeader.getValue().equals("text/event-stream")) {
```

Comment:
nit: Flip the constant to the left-hand side of the `.equals`

Reply:
Done